File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        for(string str:tokens){
            if(str=="+"){
                int a = stk.top();stk.pop();
                int b = stk.top();stk.pop();
                stk.push(b+a);
            }else if(str=="-"){
                int a = stk.top();stk.pop();
                int b = stk.top();stk.pop();
                stk.push(b-a);
            }else if(str=="/"){
                int a = stk.top();stk.pop();
                int b = stk.top();stk.pop();
                stk.push(b/a);
            }else if(str=="*"){
                int a = stk.top();stk.pop();
                int b = stk.top();stk.pop();
                stk.push(b*a);
            }else{
                stk.push(stoi(str));
            }
        }
        return stk.top();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stk = new Stack<>();
        for(int i=0; i<temperatures.length; i++){
            while(!stk.isEmpty() && temperatures[stk.peek()] < temperatures[i]){
                int ind = stk.pop();
                temperatures[ind] = i-ind;
            }
            stk.push(i);
        }
        while (!stk.isEmpty()){
            int ind = stk.pop();
            temperatures[ind]=0;
        }
        return temperatures;
    }
}
